generators/client-2/templates/src/main/webapp/app/blocks/interceptor/_notification.interceptor.ts

```diff
@@ -1,22 +1,31 @@
-import * as angular from 'angular';
+import { HttpInterceptable } from './http.interceptable';
+import { RequestOptionsArgs, Response } from '@angular/http';
+import { Observable } from 'rxjs/Observable';
 
-NotificationInterceptor.$inject = ['$q'];
+export class NotificationInterceptor extends HttpInterceptable {
 
-export function NotificationInterceptor ($q) {
-    var service = {
-        response: response
-    };
+    constructor() {
```

Comment:
That might be the case but l left it like that because  later we will need to inject the AlertService into this interceptor and l planned to do this within the constructor